File: typescript-client-sdk/src/sdk/models/operations/getcustomers.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import { Expose, Type } from "class-transformer";


export class GetCustomers200ApplicationJSONPaginationMetadata extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "has_more" })
  hasMore: boolean;

  @SpeakeasyMetadata()
  @Expose({ name: "next_cursor" })
  nextCursor: string;
}

export class GetCustomers200ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ elemType: shared.Customer })
  @Expose({ name: "data" })
  @Type(() => shared.Customer)
  data: shared.Customer[];

  @SpeakeasyMetadata()
  @Expose({ name: "pagination_metadata" })
  @Type(() => GetCustomers200ApplicationJSONPaginationMetadata)
  paginationMetadata: GetCustomers200ApplicationJSONPaginationMetadata;
}

export class GetCustomersResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getCustomers200ApplicationJSONObject?: GetCustomers200ApplicationJSON;
}

File: python-client-sdk/src/orbapi/models/operations/get_customers.py
import dataclasses
from ..shared import customer as shared_customer
from dataclasses_json import dataclass_json
from orbapi import utils
from typing import Optional


@dataclass_json
@dataclasses.dataclass
class GetCustomers200ApplicationJSONPaginationMetadata:
    has_more: bool = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('has_more') }})
    next_cursor: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('next_cursor') }})
    

@dataclass_json
@dataclasses.dataclass
class GetCustomers200ApplicationJSON:
    data: list[shared_customer.Customer] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    pagination_metadata: GetCustomers200ApplicationJSONPaginationMetadata = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('pagination_metadata') }})
    

@dataclasses.dataclass
class GetCustomersResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    get_customers_200_application_json_object: Optional[GetCustomers200ApplicationJSON] = dataclasses.field(default=None)
    